test/e2e/storage/testsuites/stress.go

```diff
@@ -141,27 +157,16 @@ func (t *stressTestSuite) DefineTests(driver TestDriver, pattern testpatterns.Te
 		l.migrationCheck.validateMigrationVolumeOpCounts()
 	}
 
-	ginkgo.It("multiple pods should access different volumes repeatedly [Slow] [Serial]", func() {
-		l := init()
-		defer func() {
-			cleanup(l)
-		}()
-
-		for i := 0; i < l.testOptions.NumPods; i++ {
-			framework.Logf("Creating resources for pod %v/%v", i, l.testOptions.NumPods-1)
-			r := CreateVolumeResource(driver, l.config, pattern, t.GetTestSuiteInfo().SupportedSizeRange)
-			l.resources = append(l.resources, r)
-			podConfig := e2epod.Config{
-				NS:           f.Namespace.Name,
-				PVCs:         []*v1.PersistentVolumeClaim{r.Pvc},
-				SeLinuxLabel: e2epv.SELinuxLabel,
-			}
-			pod, err := e2epod.MakeSecPod(&podConfig)
-			framework.ExpectNoError(err)
+	ginkgo.BeforeEach(func() {
+		init()
+		createPodsAndVolumes()
+	})
 
-			l.pods = append(l.pods, pod)
-		}
+	f.AddAfterEach("cleanup", func(f *framework.Framework, failed bool) {
```

Comment:
Do you know what the difference is between `AddAfterEach` and `ginkgo.AfterEach`?

Reply:
`AddAfterEach` appends a function to execute as part of the framework's `AfterEach`, which is registered with ginkgo via `ginkgo.AfterEach`:
https://github.com/kubernetes/kubernetes/blob/7bf9b0a3beca74dabb85f1d0034ad78f689f807c/test/e2e/framework/framework.go#L175

I expected I could replace this with a call to `ginkgo.AfterEach` and see it work, but when I tested it, it didn't run.